src/Servers/Kestrel/Core/src/Internal/Http2/FlowControl/InputFlowControl.cs

```diff
@@ -19,94 +19,125 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http2.FlowControl;
 /// <seealso href="https://datatracker.ietf.org/doc/html/rfc9113#name-flow-control"/>
 internal sealed class InputFlowControl
 {
-    private readonly int _initialWindowSize;
+    private struct FlowControlState
+    {
+        private const long AbortedBitMask = 1L << 32; // uint MaxValue + 1
+        internal long _state;
+
+        public FlowControlState(uint initialWindowSize, bool isAborted)
+        {
+            _state = initialWindowSize;
+            if (isAborted)
+            {
+                _state |= AbortedBitMask;
+            }
+        }
+
+        public uint Available => (uint)_state;
+
+        public bool IsAborted => _state > uint.MaxValue;
+    }
+
+    private readonly uint _initialWindowSize;
     private readonly int _minWindowSizeIncrement;
 
-    private FlowControl _flow;
+    private FlowControlState _flow;
     private int _pendingUpdateSize;
     private bool _windowUpdatesDisabled;
-    private readonly object _flowLock = new object();
 
     public InputFlowControl(uint initialWindowSize, uint minWindowSizeIncrement)
     {
         Debug.Assert(initialWindowSize >= minWindowSizeIncrement, "minWindowSizeIncrement is greater than the window size.");
 
-        _flow = new FlowControl(initialWindowSize);
-        _initialWindowSize = (int)initialWindowSize;
+        _flow = new FlowControlState(initialWindowSize, false);
+        _initialWindowSize = initialWindowSize;
         _minWindowSizeIncrement = (int)minWindowSizeIncrement;
     }
 
     public bool IsAvailabilityLow => _flow.Available < _minWindowSizeIncrement;
 
+    // Test hook, not participating in mutual exclusion
```

Comment:
Which mutual exclusion?

Reply:
rephrasing, but the comment makes not too much sense